### ISSUE TITLE
[4.1.0] Bug: locks initial amount after revoking and re-delegating with different STX amount

### ISSUE THREAD

**Issue by pfleigo** [blockstack/stacks-wallet#686]
I used the Stacks Wallet 4.1.0 for pooled Stacking to get into cycle #4. Here's what I did:
1. Delegated AMOUNT_A STX to Friedger Pool
2. One day later, I revoked my delegation in order to increase the amount. I waited until the revoking transaction was confirmed.
3. Then I delegated again to Friedger Pool with AMOUNT_B STX. Transaction was successful (happened one day before start of cycle #4)
4. Yesterday, when cycle #4 started, the wallet showed below my balance that **AMOUNT_A** is locked, but on the right side it says "You've delegated up to **AMOUNT_B** STX"

Checking the Stacks Explorer, it confirms that AMOUNT_A got locked. Not sure if this is a problem with the wallet or the revoking function has a bug. Or is it up to the pool provider which amount will be locked?

**Comment by markmhx** [comment]
cc @friedger in case he can help diagnose the issue here.

I suspect that he may have already stacked your AMOUNT_A for cycle #4 before your step #2 in which you revoked in order to delegate AMOUNT_B.

It also indeed in the pool provider's power to stack _less_ than the maximum delegated at any given time, although it would be a strange coincidence if he happened to lock exactly AMOUNT_A for you having been granted AMOUNT_B before stacking your STX.

**Comment by friedger** [comment]
What happend is that I fetched the information about the delegated amounts while AMOUNT_A was delegated. Then a few hours later, I used this information to lock the AMOUNT_A. Inbetween @pfleigo has changed the amount before I have locked the STX. Because @pfleigo has re-delegated before I have locked the stx it was possible to change the amount to AMOUNT_B.

In conclusion, this is expected behaviour @markmhx . I indeed had a few hours between loading the delegation data and locking the STX. As a pool admin I had to make a decision off-chain about the amount to lock, then send the corresponding transaction. There will always be a small time gap during that users can change the amount.

Possible improvements would be https://github.com/clarity-lang/reference/issues/30 This would help to do the decision on-chain during the locking transaction.
